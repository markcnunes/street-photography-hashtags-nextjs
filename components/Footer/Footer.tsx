import styled from '@emotion/styled';
import Link from 'next/link';

const StyledFooter = styled('footer')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  alignItems: 'center',
  width: '100%',
  padding: 20,
  fontSize: theme.fontSizes[1],
  color: theme.colors.white,
  '@media (min-width: 780px)': {
    flexDirection: 'row',
  },
}));

const StyledAuthor = styled.div(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  marginBottom: 20,
  a: {
    color: theme.colors.white,
    marginLeft: 5,
    '&:hover': {
      color: theme.colors.white,
    },
  },
  '@media (min-width: 780px)': {
    marginBottom: 0,
  },
}));

const StyledLinks = styled.div(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flex: '1 0 0',
  marginBottom: 20,
  a: {
    color: theme.colors.white,
    margin: '0 10px',
  },
  '@media (min-width: 780px)': {
    marginBottom: 0,
  },
}));

const StyleCta = styled.a(({ theme }) => ({
  fontSize: theme.fontSizes[1],
  color: 'currentColor',
  border: '1px solid currentColor',
  padding: '8px 16px',
  borderRadius: 5,
  textDecoration: 'none',
}));

const Footer: React.FC = () => {
  return (
    <StyledFooter>
      <StyledAuthor>
        Created by
        <a
          href='https://www.instagram.com/markcnunes/'
          title="Mark's website"
          target='_blank'
          rel='noopener noreferrer'
        >
          Mark Claus Nunes
        </a>
      </StyledAuthor>
      <StyledLinks>
        <Link href={`/privacy`}>Privacy</Link>
        <Link href={`/terms`}>Terms of use</Link>
      </StyledLinks>
      <Link href='/contribution'>
        <StyleCta>Contributions are welcomed!</StyleCta>
      </Link>
    </StyledFooter>
  );
};

export default Footer;
