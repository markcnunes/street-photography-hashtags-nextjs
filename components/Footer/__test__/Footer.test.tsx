import React from 'react';
import { render } from 'test/setupTests';

import Footer from '../Footer';

describe('<Footer />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
