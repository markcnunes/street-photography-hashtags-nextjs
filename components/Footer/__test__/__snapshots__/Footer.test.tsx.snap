// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 20px;
  font-size: 14px;
  color: #fff;
}

@media (min-width: 780px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.emotion-2 a {
  color: #fff;
  margin-left: 5px;
}

.emotion-2 a:hover {
  color: #fff;
}

@media (min-width: 780px) {
  .emotion-2 {
    margin-bottom: 0;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  margin-bottom: 20px;
}

.emotion-4 a {
  color: #fff;
  margin: 0 10px;
}

@media (min-width: 780px) {
  .emotion-4 {
    margin-bottom: 0;
  }
}

.emotion-6 {
  font-size: 14px;
  color: currentColor;
  border: 1px solid currentColor;
  padding: 8px 16px;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<footer
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      Created by
      <a
        href="https://www.instagram.com/markcnunes/"
        rel="noopener noreferrer"
        target="_blank"
        title="Mark's website"
      >
        Mark Claus Nunes
      </a>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      <a
        href="/privacy"
      >
        Privacy
      </a>
      <a
        href="/terms"
      >
        Terms of use
      </a>
    </div>
    <a
      class="emotion-6 emotion-7"
    >
      Contributions are welcomed!
    </a>
  </footer>
</DocumentFragment>
`;
